mbuild/compound.py

```diff
@@ -2235,6 +2346,208 @@ def _energy_minimize_openbabel(
+                        p1 = fixed_temp[0]
+                        dims = [True, True, True]
+                else:
+                    p1 = fixed_temp
+                    dims = [True, True, True]
+
+                all_true = False
+                if dims[0] == dims[1] == dims[2] == True:
+                    all_true = True
+
+                p1_in_compound = False
+                for successor in self.successors():
+                    if id(p1) == id(successor):
+                        p1_in_compound = True
+                if id(p1) == id(self):
+                    p1_in_compound = True
+
+                if p1_in_compound == False:
+                    raise MBuildError(
+                        f"{p1} is not a member of Compound {self}."
+                    )
+
+                if len(p1.children) == 0:
+                    index = 1
+                    for particle in self.particles():
+                        if id(p1) == id(particle):
+                            pid = index
+                            if all_true == True:
+                                ob_constraints.AddAtomConstraint(pid)
+                            elif dims[0] == True:
+                                ob_constraints.AddAtomXConstraint(pid)
+                            elif dims[1] == True:
+                                ob_constraints.AddAtomYConstraint(pid)
+                            elif dims[2] == True:
+                                ob_constraints.AddAtomZConstraint(pid)
+                        index += 1
+                else:
+                    for successor in p1.successors():
+                        index = 1
+                        for particle in self.particles():
+                            if id(successor) == id(particle):
+                                pid = index
+                                if all_true == True:
+                                    ob_constraints.AddAtomConstraint(pid)
+                                elif dims[0] == True:
+                                    ob_constraints.AddAtomXConstraint(pid)
+                                elif dims[1] == True:
+                                    ob_constraints.AddAtomYConstraint(pid)
+                                elif dims[2] == True:
+                                    ob_constraints.AddAtomZConstraint(pid)
+                            index += 1
+
+        if ignore_compounds is not None:
+            temp1 = np.array(ignore_compounds)
+            if len(temp1.shape) == 2:
+                ignore_compounds = list(temp1.reshape(-1))
+            for ignore in ignore_compounds:
+                p1 = ignore
+                p1_in_compound = False
+                for successor in self.successors():
+                    if id(p1) == id(successor):
+                        p1_in_compound = True
+                if id(p1) == id(self):
+                    p1_in_compound = True
+                if p1_in_compound == False:
```

Comment:
A thought that might clean up this code some would be to hide away all of these checks into a `validation` function that would get run at the top of any of these calls. Might end up making this code a little easier to follow, since it's a lot of lines in total.

Reply:
I'm inclined to just leave it the way it is at this point the way it is. It's not really something very general that would be recycled.